clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -79,14 +79,15 @@
                                            + " durability of records that are sent. The following settings are allowed: "
                                            + " <ul>"
                                            + " <li><code>acks=0</code> If set to zero then the producer will not wait for any acknowledgment from the"
-                                           + " server at all. The record will be immediately added to the socket buffer and considered sent. No guarantee can be"
+                                           + " server at all. The record will be immediately added to the socket buffer and considered sent. There is no guarantee"
+                                           + " that the socket buffer will be flushed prior to the shutdown of the producer JVM.  No guarantee can be"
```

Comment:
@ijuma This is the issue I was planning to work on under KAFKA-3703. Non-blocking close does not flush outgoing buffers. So if the JVM exits, data is lost when acks=0. Having tried out some fixes unsuccessfully, it may be better to document this limitation better.